ext/GraphPPLGraphVizExt.jl

```diff
@@ -665,16 +634,19 @@ end
 # # Returns:
 # - `String`: A DOT format string that can be used to generate a GraphViz visualization.
 # """
-function _generate_dot(;
-    model_graph::GraphPPL.Model,
-    strategy::TraversalStrategy,
-    font_size::Int,
+function GraphViz.load(model_graph::GraphPPL.Model;
```

Comment:
I have attempted to call this method in a version of the v2 test file that I wrote as follows:

```julia
push!(LOAD_PATH, joinpath(@__DIR__, "../ext"))

using GraphPPLGraphVizExt: GraphViz


## CREATE AN RXINFER.JL MODEL:
# GraphPPL.jl export `@model` macro for model specification
# It accepts a regular Julia function and builds an FFG under the hood
@model function coin_model(y, a, b)
    # We endow θ parameter of our model with some prior
    θ ~ Beta(a, b)
    # or, in this particular case, the `Uniform(0.0, 1.0)` prior also works:
    # θ ~ Uniform(0.0, 1.0)

    # We assume that outcome of each coin flip is governed by the Bernoulli distribution
    for i in eachindex(y)
        y[i] ~ Bernoulli(θ)
    end
end

# condition the model on some observed data
conditioned = coin_model(a = 2.0, b = 7.0) | (y = [ true, false, true ], )

# `Create` the actual graph of the model conditioned on the data
rxi_model = RxInfer.create_model(conditioned)

gppl_model = RxInfer.getmodel(rxi_model)


gen_dot_result_coin_simple = GraphViz.load(
    gppl_model,
    strategy = :simple,
    font_size = 12,
    edge_length = 1.0,
    layout = "neato",
    overlap = false,
    width = 10.0,
    height = 10.0,
    show = false
)
```

As yet, my use of `GraphViz.load` is resulting in a persistent failure to correctly dispatch on the overloaded method: 

```BASH
$ julia test/visualization_tests_233_v2.jl
ERROR: LoadError: MethodError: no method matching var"#load#3"(::Symbol, ::Int64, ::Float64, ::String, ::Bool, ::Float64, ::Float64, ::Bool, ::Nothing, ::typeof(GraphViz.load), ::GraphPPL.Model{MetaGraph{Int64, SimpleGraph{Int64}, GraphPPL.NodeLabel, GraphPPL.NodeData, GraphPPL.EdgeLabel, GraphPPL.Context, MetaGraphsNext.var"#4#8", Float64}, GraphPPL.PluginsCollection{Tuple{}}, RxInfer.ReactiveMPGraphPPLBackend{Static.False}})
The function `#load#3` exists, but no method is defined for this combination of argument types.

Closest candidates are:
  var"#load#3"(::Symbol, ::Int64, ::Float64, ::String, ::Bool, ::Float64, ::Float64, ::Bool, ::String, ::typeof(GraphViz.load), ::GraphPPL.Model)
   @ GraphPPLGraphVizExt ~/Desktop/GraphPPL.jl/ext/GraphPPLGraphVizExt.jl:637

Stacktrace:
 [1] load(model_graph::GraphPPL.Model{MetaGraph{Int64, SimpleGraph{Int64}, GraphPPL.NodeLabel, GraphPPL.NodeData, GraphPPL.EdgeLabel, GraphPPL.Context, MetaGraphsNext.var"#4#8", Float64}, GraphPPL.PluginsCollection{Tuple{}}, RxInfer.ReactiveMPGraphPPLBackend{Static.False}})
   @ GraphPPLGraphVizExt ~/Desktop/GraphPPL.jl/ext/GraphPPLGraphVizExt.jl:637
 [2] top-level scope
   @ ~/Desktop/GraphPPL.jl/test/visualization_tests_233_v2.jl:51
```

Perhaps I have just done something silly but I can't seem to shake this error, no matter how I re-define the positionality of the arguments.

Reply:
*Update*: Ok its my mistake, I pushed the fix, the type for the `save_to` variable was wrong.

I'm not sure about those lines
```
push!(LOAD_PATH, joinpath(@__DIR__, "../ext"))

using GraphPPLGraphVizExt: GraphViz
```
You should simply do
```
using GraphViz
```
and the package should be installed on your system. Otherwise the extension won't load properly. Does that fix the error?

You error however says something weird, so it definitely sees the method here in the stacktrace (I assume the extension loaded?)
```julia
load(model_graph::GraphPPL.Model{MetaGraph{Int64, SimpleGraph{Int64}, GraphPPL.NodeLabel, GraphPPL.NodeData, GraphPPL.EdgeLabel, GraphPPL.Context, MetaGraphsNext.var"#4#8", Float64}, GraphPPL.PluginsCollection{Tuple{}}, RxInfer.ReactiveMPGraphPPLBackend{Static.False}})
   @ GraphPPLGraphVizExt ~/Desktop/GraphPPL.jl/ext/GraphPPLGraphVizExt.jl:637
```

Reply:
Thanks for the clarification @bvdmitri.

I have two things to discuss. 

# First Thing:

You seem to be saying that I can/should simply replace the following lines in `test/visualization_tests_233_v2.jl`: 

```julia
push!(LOAD_PATH, joinpath(@__DIR__, "../ext"))
using GraphPPLGraphVizExt: GraphViz
```

with the simpler: `using GraphViz` . I did try this, it results in:

```BASH
$ julia test/visualization_tests_233_v2.jl
ERROR: LoadError: MethodError: no method matching load(::GraphPPL.Model{MetaGraph{Int64, SimpleGraph{Int64}, GraphPPL.NodeLabel, GraphPPL.NodeData, GraphPPL.EdgeLabel, GraphPPL.Context, MetaGraphsNext.var"#4#8", Float64}, GraphPPL.PluginsCollection{Tuple{}}, RxInfer.ReactiveMPGraphPPLBackend{Static.False}}; strategy::Symbol)
The function `load` exists, but no method is defined for this combination of argument types.

Closest candidates are:
  load(::IO) got unsupported keyword argument "strategy"
   @ GraphViz ~/.julia/packages/GraphViz/IsUMl/src/GraphViz.jl:90
  load(::FileIO.File{FileIO.DataFormat{:DOT}}) got unsupported keyword argument "strategy"
   @ GraphViz ~/.julia/packages/GraphViz/IsUMl/src/GraphViz.jl:89

Stacktrace:
 [1] top-level scope
   @ ~/Desktop/GraphPPL.jl/test/visualization_tests_233_v2.jl:50
in expression starting at / ... /test/visualization_tests_233_v2.jl:50
```

Thus it seems that Julia is unable to find the overloaded version of `GraphViz.load` if I don't explicitly use 

```julia 
using GraphPPLGraphVizExt: GraphViz
``` 

to specify that I want my overlaoded version from `GraphPPLGraphVizExt`. In order to do _this_, I think I need to append the extension to Julia's `LOAD_PATH` variable. That is why I have:

```julia
push!(LOAD_PATH, joinpath(@__DIR__, "../ext"))
```

My understanding is that it is necessary to append the `ext` directory to Julia's current list of available directories in the `LOAD_PATH` variable. Perhaps there is a much better/cleaner way to do this.  

# Second Thing:

When I attempt to run my existing test script:

```julia
using GraphPPL
using Test
using RxInfer
using Distributions
using Random
using Graphs
using MetaGraphsNext
using Dictionaries
using Cairo
using Fontconfig
using Compose
using GraphPlot

push!(LOAD_PATH, joinpath(@__DIR__, "../ext"))

using GraphPPLGraphVizExt: GraphViz

## CREATE AN RXINFER.JL MODEL:
# GraphPPL.jl export `@model` macro for model specification
# It accepts a regular Julia function and builds an FFG under the hood
@model function coin_model(y, a, b)
    # We endow θ parameter of our model with some prior
    θ ~ Beta(a, b)

    # We assume that outcome of each coin flip is governed by the Bernoulli distribution
    for i in eachindex(y)
        y[i] ~ Bernoulli(θ)
    end
end

# condition the model on some observed data
conditioned = coin_model(a = 2.0, b = 7.0) | (y = [ true, false, true ], )

# `Create` the actual graph of the model conditioned on the data
rxi_model = RxInfer.create_model(conditioned)

gppl_model = RxInfer.getmodel(rxi_model)

# generate visualization:
gen_dot_result_coin_simple = GraphViz.load(
    gppl_model, 
    strategy = :simple
)

println(gen_dot_result_coin_simple)
```

Julia can't seem to find the `prettyname` function:

```BASH
$ julia test/visualization_tests_233_v2.jl
ERROR: LoadError: UndefVarError: `prettyname` not defined in `GraphPPL`
Stacktrace:
 [1] get_displayed_label(properties::GraphPPL.FactorNodeProperties{GraphPPL.NodeData})
   @ GraphPPLGraphVizExt ~/Desktop/GraphPPL.jl/ext/GraphPPLGraphVizExt.jl:377
 [2] add_nodes!(io_buffer::IOBuffer, model_graph::GraphPPL.Model{MetaGraph{Int64, SimpleGraph{Int64}, GraphPPL.NodeLabel, GraphPPL.NodeData, GraphPPL.EdgeLabel, GraphPPL.Context, MetaGraphsNext.var"#4#8", Float64}, GraphPPL.PluginsCollection{Tuple{}}, RxInfer.ReactiveMPGraphPPLBackend{Static.False}}, global_namespace_dict::Dict{Int64, Dict{Symbol, Any}}, ::GraphPPLGraphVizExt.SimpleIteration)
   @ GraphPPLGraphVizExt ~/Desktop/GraphPPL.jl/ext/GraphPPLGraphVizExt.jl:423
 [3] load(model_graph::GraphPPL.Model{MetaGraph{Int64, SimpleGraph{Int64}, GraphPPL.NodeLabel, GraphPPL.NodeData, GraphPPL.EdgeLabel, GraphPPL.Context, MetaGraphsNext.var"#4#8", Float64}, GraphPPL.PluginsCollection{Tuple{}}, RxInfer.ReactiveMPGraphPPLBackend{Static.False}}; strategy::Symbol, font_size::Int64, edge_length::Float64, layout::String, overlap::Bool, width::Float64, height::Float64, show::Bool, save_to::Nothing)
   @ GraphPPLGraphVizExt ~/Desktop/GraphPPL.jl/ext/GraphPPLGraphVizExt.jl:662
 [4] top-level scope
   @ ~/ ... /test/visualization_tests_233_v2.jl:48
in expression starting at / ... /test/visualization_tests_233_v2.jl:48
```

I have tried all kinds of things to resolve this issue, none have worked. 

Perhaps I am systematically misunderstanding how Julia loads various modules and files for actual execution and this is the reason why I'm having all this difficulty. That is my guess! I apologise for any such deficiency on my behalf.

I did actually get an overloaded version of `GraphViz.load` working in my own branch (minus the `prettyname`) so I'm not sure why there are all these difficulties here. 

I hope this is enough to go on for general comments and suggestions.